erpnext/accounts/doctype/coupon_code/coupon_code.js

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Coupon Code', {
+	refresh: function(frm) {
```

Comment:
use dashboard

Reply:
Link between two docs is-- 'Coupon Code'.pricing_rule=='Pricing Rule'.title.
Hence dashboard is not used.